aws/client.py

```diff
@@ -202,7 +202,10 @@ def extract_key(self, key, default=None):
                 keyed_result = result[key]
                 if isinstance(keyed_result, list):
                     for item in keyed_result:
-                        item['__pytest_meta'] = result['__pytest_meta']
+                        # TODO: Is there a bug in usage (IAM resources) causing it
+                        # to sometimes be a string?
```

Comment:
Oh I think I know what causes this. The "inline" IAM policy methods return a list of policy names rather than a list of dictionaries. (http://botocore.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.list_user_policies)